File: command/java/02/Main.java
import java.util.function.Consumer;

// Classe base para as operações (comandos)
interface Command extends Consumer<Void> {
}

// Classe concreta para a operação "Adicionar"
class Cmd1Command implements Command {
	private int x;
	private int y;

	public Cmd1Command(int x, int y) {
		this.x = x;
		this.y = y;
	}

	public void accept(Void v) {
		System.out.println(x + y);
	}
}

// Classe que armazena e executa um comando
class Invoke {
	private Command command;

	public void setCommand(Command command) {
		this.command = command;
	}

	public void executeCommand() {
		command.accept(null);
	}
}

public class Main {
	public static void main(String[] args) {
    Invoke invoke = new Invoke();

    // Define e executa o comando "Adicionar"
    invoke.setCommand(new Cmd1Command(10, 5));
    invoke.executeCommand(); // Imprime 15
  }

File: command/python/02/main.py
from abc import ABC, abstractmethod

# Classe base para as operações (comandos)


class Command(ABC):
    @abstractmethod
    def execute(self):
        pass

# Classe concreta para a operação "Adicionar"
class AddCommand(Command):
    def __init__(self, x, y):
        self.x = x
        self.y = y

    def execute(self):
        print(self.x + self.y)

# Classe concreta para a operação "Subtrair"


class SubtractCommand(Command):
    def __init__(self, x, y):
        self.x = x
        self.y = y

    def execute(self):
        print(self.x - self.y)

# Classe que armazena e executa um comando


class Calculator:
    def __init__(self):
        self.command = None

    def set_command(self, command):
        self.command = command

    def execute_command(self):
        self.command.execute()


def main():
    calculator = Calculator()

    # Define e executa o comando "Adicionar"
    calculator.set_command(AddCommand(10, 5))
    calculator.execute_command()  # Imprime 15

    # Define e executa o comando "Subtrair"
    calculator.set_command(SubtractCommand(10, 5))
    calculator.execute_command()  # Imprime 5


if __name__ == "__main__":
    main()
